### PR TITLE
Add "Deploying your Docs" page to the User Guide.

### PR DESCRIPTION
**Pull request by waylan**
Addresses #477.

Note that I did not provide a separate section for Amazon S3. I started to but realized that the instructions are basically the same as other hosts: (1) build your site, (2) upload the contents of the `site` dir. Therefore, I included it in the "other" section. The other hosts listed in the "other" section are all from [this list] (I omitted the hosts which I could not locate any documentation for). I would expect that more specific per-host instructions could be provided by users in a publicly editable wiki.

Any and all feedback is welcome.

[this list]: http://lifehacker.com/5124856/most-popular-reliable-and-affordable-web-hosts

### CODE REVIEW

**Comment by d0ugal** [comment]
Woah, this looks great! I just skimmed it so far, will review in a bit.

File: docs/user-guide/deploying-your-docs.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+your hosts' file system. 
+
+[ftp]: https://en.wikipedia.org/wiki/File_Transfer_Protocol
+[ssh]: https://en.wikipedia.org/wiki/Secure_Shell
+[scp]: https://en.wikipedia.org/wiki/Secure_copy
+
+See your host's documentation for specifics. You will likely want to search
+their documentation for "ftp" or "uploading site". A list of instructions for a
+few popular hosts can be found below. MkDocs does not recommend or endorse any
+hosting providers.
```
Comment (d0ugal) [inline docs/user-guide/deploying-your-docs.md]: While it is good to state this, it still concerns me somewhat.

File: docs/user-guide/deploying-your-docs.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+* [Amazon S3](http://docs.aws.amazon.com/AmazonS3/latest/dev/WebsiteHosting.html)
+* [Dreamhost](http://wiki.dreamhost.com/Uploading_your_site)
+* [Blue Host](https://my.bluehost.com/cgi/help/upload-site)
+* [1&1](http://help.1and1.com/hosting-c37630/webspace-and-access-c85098/ftp-c37631/transfer-files-using-ftp-a792530.html)
+* [A Small Orange](https://kb.asmallorange.com/customer/portal/articles/1603088-upload-files-to-your-server-or-website)
+* [GoDaddy](https://support.godaddy.com/help/article/96/upload-files-to-your-website-ftp)
```
Comment (d0ugal) [inline docs/user-guide/deploying-your-docs.md]: We should remove GoDaddy at least. Search for `GoDaddy sexism` to see some of my objections.

**Comment by d0ugal** [comment]
This looks great, some concerns about the very end but otherwise I'd be happy to merge this.

File: docs/user-guide/deploying-your-docs.md (hunk 2 above)
Comment (waylan) [inline docs/user-guide/deploying-your-docs.md]: I almost left them off for that reason - but decided to not give preferential treatment to anyone. That said, this is your project and I'm okay with whatever you want. I'll remove it.

**Comment by waylan** [comment]
I removed the list as per your request. However, the list is still in the commit history if anyone wants to find it later.

**Comment by d0ugal** [comment]
Looks great, thanks! Tested locally, so I wont wait for CI :)

**Comment by d0ugal** [comment]
I just noticed I closed this and didn't merge - sorry about that. I got really confused when I couldn't find the commit!

**Pull request merged**